### PR TITLE
Clean up "Writing consistent snapshots" section

### LINKED ISSUE #184: should repository make unversioned root metadata available or not?
small inconsistency with regards to root.json versions: _6.2.1. Writing consistent snapshots_ states:

> an implementation must write both root.json and version_number.root.json because it is possible to download root metadata both with and without known version numbers.

It is _not_ in fact documented that client could download root without a version (client workflow only downloads versioned roots). I can see how a Trust-On-First-Use (TOFU) client initialization could use this but that is not defined in the spec.

I guess either
* TOFU initialization could be mentioned as a possibility (even if not recommended)
* the requirement to offer unversioned root.json should not be there

### PR DESCRIPTION
**Pull request by joshuagl**
* Match how metadata and target filenames are referred to in the detailed
  client workflow, i.e. filename.ext -> FILENAME.EXT
* Remove suggestion to write a version of the root metadata without a
  version prefix. We only ever look for version-prefixed version of root
  metadata.

Fixes #184

### CODE REVIEW

File: tuf-spec.md
Comment (mnm678) [inline tuf-spec.md:1603]: per RFC 2119, I think this should be a MUST

File: tuf-spec.md
Comment (mnm678) [inline tuf-spec.md:1594]: In general I like `DIGEST` better than `HASH`, but I know that we use `HASH` elsewhere in the spec. No need to change it unless others feel strongly.

File: tuf-spec.md
Comment (mnm678) [inline tuf-spec.md:1588]: I think this and the below paragraph should also be MUSTs.

File: tuf-spec.md
Comment (joshuagl) [reply tuf-spec.md:1603]: I will make that change. I was thinking of SHOULD in the context of #185, but I should handle that separately.

File: tuf-spec.md
Comment (joshuagl) [reply tuf-spec.md:1594]: I used `HASH` because that's what we use in the the [5.7.3](https://theupdateframework.github.io/specification/v1.0.25/#fetch-target) of the detailed client workflow and the name of the attribute in metadata is 'hashes'.

File: tuf-spec.md
Comment (joshuagl) [reply tuf-spec.md]: I've added quite a few more RFC 2119 keywords in 6c54aa8 and cleaned up other paragraphs in the section.

**Comment by joshuagl** [comment]
Thanks for the review @mnm678, I've addressed your comments and clarified some more of the text in the "Writing consistent snapshots" section. Please take a look.

**Review by mnm678 — APPROVED** [review]

**Comment by joshuagl** [comment]
@lukpueh @trishankatdatadog could we get a second review here? Thank you!

File: tuf-spec.md
Comment (lukpueh) [inline tuf-spec.md:1607]: Just took me a while to parse the sentence. For me it would have been helpful to read e.g. like so

```suggestion
version numbers.  It is OPTIONAL for an implementation to write identical copies of timestamp.EXT to the respective VERSION_NUMBER.timestamp.EXT for record-keeping purposes.
```

**Review by lukpueh — APPROVED** [review]
Great set of clarifications, @joshuagl! I added one minor suggestion. Feel free to ignore if you think the phrase is obvious without it.

**Comment by joshuagl** [comment]
Thanks for the review both. I committed Lukas' change, which dismissed existing reviews. Please take another look @mnm678 @lukpueh

**Review by lukpueh — APPROVED** [review]

**Review by mnm678 — DISMISSED** [review]

File: tuf-spec.md
Comment (trishankatdatadog) [inline tuf-spec.md:1606]:
```suggestion
version numbers.  It is OPTIONAL for an implementation to write an identical copy
```

**Review by trishankatdatadog — APPROVED** [review]

File: tuf-spec.md
Comment (joshuagl) [reply tuf-spec.md]: 💯 thanks

**Comment by joshuagl** [comment]
Thanks for the suggestion Trishank. I've committed it and dismissed all reviews. Please take another look all @mnm678 @lukpueh @trishankatdatadog

**Review by lukpueh — APPROVED** [review]
Still LGTM!

**Review by mnm678 — APPROVED** [review]

**Pull request merged**
